### PR TITLE
KHWE-14: Implement the simple router

### PR DESCRIPTION
**Pull request by lubovgribiniyk**

### CODE REVIEW

File: src/components/Main/Main.js [hunk 1]
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
...
     }
+
+    if (pathname === "about") {
+      this.about.mount();
+      return;
+    }
+
+    const urlParts = pathname.split("/");
+
+    if (urlParts[0] === "songs" && urlParts[1] && urlParts.length === 2) {
```
Comment (ihor4568) [inline src/components/Main/Main.js]: please contact Ivan - we have agreed on different path format for share url

File: src/components/NotFound/NotFound.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1 class="not-found mdc-typography--headline4">404: Page Not Found</h1>
```
Comment (ihor4568) [inline src/components/NotFound/NotFound.html]: please make text `Not Found`

Comment (ihor4568) [reply src/components/NotFound/NotFound.html]: also make its color gray. And display it at the center of the screen

File: src/components/Main/Main.js [hunk 3]
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
 
   addEventListeners() {
     this.sidebarList.addEventListener("click", this.handleListClick.bind(this));
```
Comment (ghost) [inline src/components/Main/Main.js:35]: bind handleListClick function in the constructor once. you should not call bind function every time event triggers

File: src/components/Main/Main.js [hunk 4]
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
...
+    window.addEventListener("popstate", this.handleStatePath.bind(this));
```
Comment (ghost) [inline src/components/Main/Main.js:36]: same

File: src/components/Main/Main.js (hunk 3 above)
Comment (lubovgribiniyk) [reply src/components/Main/Main.js:35]: `bind` is called only when function `addEventListeners` is called and bind will return new function which will be set us as an event handler for `click` event.
In our case `addEventListeners` will be called only once. So `bind` will be called only once.

File: src/components/Main/Main.js [hunk 5]
```diff
@@ -30,15 +31,55 @@ export class MainComponent {
...
   }
 
   handleListClick(e) {
     e.preventDefault();
-    if (e.target.closest(".main__about-link")) {
-      this.about.mount();
-    } else if (e.target.closest(".main__albums-link")) {
+    const { target } = e;
```
Comment (ghost) [inline src/components/Main/Main.js:41]: you use it once in your block code, so i am not sure it is necessary to destruct the object. why not simply access the appropriate property as is through the dot notation?

File: src/components/Main/Main.js (hunk 3 above)
Comment (ihor4568) [reply src/components/Main/Main.js:35]: actually not - it binds one time when addEventListeners function is being called

File: src/components/Main/Main.js (hunk 5 above)
Comment (ihor4568) [reply src/components/Main/Main.js:41]: not sure if it makes code any worse

**Comment by ihor4568** [comment]
could you please also center not found vertically?

image

**Review by gamesminer — APPROVED** [review]

**Pull request merged**